src/main/java/armeria/TracingFactory.java

```diff
@@ -25,11 +28,19 @@ static Sender sender() {
   }
 
   /** Configuration for how to buffer spans into messages for Zipkin */
-  static AsyncReporter<Span> spanReporter() {
-    final AsyncReporter<Span> result = AsyncReporter.create(sender());
-    // make sure spans are reported on shutdown
-    Runtime.getRuntime().addShutdownHook(new Thread(result::close));
-    return result;
+  static AsyncReporter<Span> spanReporter(Sender sender) {
+    final AsyncReporter<Span> spanReporter = AsyncReporter.create(sender);
+
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      spanReporter.close(); // Make sure spans are reported on shutdown
+      try {
+        sender.close(); // Release any network resources used to send spans
```

Comment:
note technically there is nothing done in URLConnection. However, this is important if someone switched to kafka